### PR TITLE
Multiple Dictionary Requests Fix

### LINKED ISSUE #246: Multiple Dictionary Requests if viewing deep dictionary item on load
#### Summary

This issue is two-fold.

1. Loading the page while viewing a deeply nested dictionary item is causing multiple requests back to back for the dictionary (from the router).
2. There is an issue with our logic regarding async/await and the dictionary promise.

#### Expected vs Current Behavior

The minimum number of requests should be made (2 at this point).

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken? *Maybe*
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?

#### Steps to Reproduce

1. view an item deeply nested (3 - 4 items deep) in a dictionary.
3. look at calls in the network tab
4. many duplicated calls are being made

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #246

### Describe your changes:

First off, thank you @shefalijoshi for troubleshooting this!

Reworked the `getTelemetryDictionary` method and the `fetchTelemetryDictionary` method by using a promise.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [x] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/providers/object-provider.js [hunk 1]
```diff
@@ -176,44 +176,40 @@ export default class YamcsObjectProvider {
     async #getTelemetryDictionary() {
         if (this.dictionaryLoaded) {
             return this.dictionary;
```
Comment (shefalijoshi) [inline src/providers/object-provider.js:178]: Where is this.dictionary being assigned? Maybe missed that?

File: src/providers/object-provider.js [hunk 2]
```diff
@@ -176,44 +176,40 @@ export default class YamcsObjectProvider {
...
-        this.dictionaryLoaded = true;
-        this.roleStatusTelemetry.dictionaryLoadComplete();
+        this.dictionaryPromise = this.#fetchTelemetryDictionary(this.url, this.instance, this.folderName)
+            .finally(() => {
+                this.dictionaryLoaded = true;
+                this.roleStatusTelemetry.dictionaryLoadComplete();
+            });
 
-        return dictionary;
+        return this.dictionaryPromise;
```
Comment (shefalijoshi) [inline src/providers/object-provider.js:189]: Can we have this always return the promise?

**Review by shefalijoshi — CHANGES_REQUESTED** [review]

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Can you add or update this suite to detect the multiple requests https://github.com/akhenry/openmct-yamcs/blob/master/tests/e2e/yamcs/network.e2e.spec.js

File: src/providers/object-provider.js (hunk 1 above)
Comment (jvigliotta) [reply src/providers/object-provider.js:178]: https://github.com/akhenry/openmct-yamcs/blob/dac68ba6d18f9769fa4d971bf7d84476386551c2/src/providers/object-provider.js#L209 this triggers that.

File: src/providers/object-provider.js (hunk 2 above)
Comment (jvigliotta) [reply src/providers/object-provider.js:189]: This shouldn't be an issue, the way the logic is setup, it should only every be called one time. Once `this.dictionaryLoaded` is set, it's never changed.

**Comment by scottbell** [comment]
This is great @jvigliotta! I think we can simplify the logic a bit and get rid of the `dictionaryLoaded` boolean. I've got a proposal branch here (along with a network e2e test) here:
https://github.com/akhenry/openmct-yamcs/compare/dupe-dictionary-fix...dupe-dictionary-fix-with-test?expand=1
Take it or leave it! 😀

**Comment by jvigliotta** [comment]
> This is great @jvigliotta! I think we can simplify the logic a bit and get rid of the `dictionaryLoaded` boolean. I've got a proposal branch here (along with a network e2e test) here: https://github.com/akhenry/openmct-yamcs/compare/dupe-dictionary-fix...dupe-dictionary-fix-with-test?expand=1 Take it or leave it! 😀

That's perfect! Thanks I'll merge it in. Thanks!

**Review by scottbell — APPROVED** [review]
LGTM!

File: tests/e2e/yamcs/network.e2e.spec.js [hunk 3]
```diff
@@ -68,6 +68,16 @@ test.describe("Quickstart network requests @yamcs", () => {
         // no second request (for limits) should be made
         expect(filterNonFetchRequests(networkRequests).length).toBe(1);
 
+        networkRequests = [];
```
Comment (unlikelyzero) [inline tests/e2e/yamcs/network.e2e.spec.js:71]: this looks good. I think we may want to move to discrete tests in the future

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**